File: r/tests/testthat/test-env.R
test_that("default uses prod host", {
  old <- Sys.getenv("BB_ENV", unset = NA)
  on.exit({
    if (is.na(old)) Sys.unsetenv("BB_ENV") else Sys.setenv(BB_ENV = old)
    bedrockbio:::.onLoad(NULL, NULL)
  })
  Sys.unsetenv("BB_ENV")
  bedrockbio:::.onLoad(NULL, NULL)
  pkg <- bedrockbio:::pkg
  expect_equal(pkg$manifest_url, "https://datasets.bedrock.bio/manifest.json")
})

test_that("BB_ENV=dev uses dev host", {
  old <- Sys.getenv("BB_ENV", unset = NA)
  on.exit({
    if (is.na(old)) Sys.unsetenv("BB_ENV") else Sys.setenv(BB_ENV = old)
    bedrockbio:::.onLoad(NULL, NULL)
  })
  Sys.setenv(BB_ENV = "dev")
  bedrockbio:::.onLoad(NULL, NULL)
  pkg <- bedrockbio:::pkg
  expect_equal(pkg$manifest_url, "https://datasets-dev.bedrock.bio/manifest.json")
})

test_that("reset re-resolves host from BB_ENV", {
  old <- Sys.getenv("BB_ENV", unset = NA)
  on.exit({
    if (is.na(old)) Sys.unsetenv("BB_ENV") else Sys.setenv(BB_ENV = old)
    bedrockbio:::reset()
  })
  Sys.setenv(BB_ENV = "dev")
  bedrockbio:::reset()
  expect_equal(
    bedrockbio:::pkg$manifest_url,
    "https://datasets-dev.bedrock.bio/manifest.json"
  )
})

File: python/tests/test_env.py
from bedrock_bio.config import config


def test_default_uses_prod_host(monkeypatch):
    monkeypatch.delenv("BB_ENV", raising=False)
    assert config.base_url == "https://datasets.bedrock.bio"
    assert config.manifest_url == "https://datasets.bedrock.bio/manifest.json"


def test_dev_uses_dev_host(monkeypatch):
    monkeypatch.setenv("BB_ENV", "dev")
    assert config.base_url == "https://datasets-dev.bedrock.bio"
    assert config.manifest_url == "https://datasets-dev.bedrock.bio/manifest.json"
